magma/wire_container.py

```diff
@@ -191,3 +191,53 @@ def wire(self, o, debug_info):
         self._wire.connect(o._wire, debug_info)
         self.debug_info = debug_info
         o.debug_info = debug_info
+
+
+class WireableWithChildren(Wireable):
+    def _has_elaborated_children(self):
```

Comment:
let's decorate these with `@abc.abstractmethod`

unfortunately, without inheriting from `abc.ABC` also, it will be a no-op, but at-least it serves as good documentation.